File: copilot/jsp/securable/src/main/java/com/loosenotes/model/ShareLink.java
package com.loosenotes.model;

/**
 * A unique token-based share link for a note.
 * Anyone with the token can view the note without authentication.
 */
public class ShareLink {

    private long id;
    private long noteId;
    private String token;
    private long createdAt;

    public ShareLink() {}

    public long getId() { return id; }
    public void setId(long id) { this.id = id; }

    public long getNoteId() { return noteId; }
    public void setNoteId(long noteId) { this.noteId = noteId; }

    /** Cryptographically random 64-hex-char token. */
    public String getToken() { return token; }
    public void setToken(String token) { this.token = token; }

    public long getCreatedAt() { return createdAt; }
    public void setCreatedAt(long createdAt) { this.createdAt = createdAt; }
}

File: claude/aspnet/rawdog/Models/ShareLink.cs
namespace LooseNotes.Models;

public class ShareLink
{
    public int Id { get; set; }
    public string Token { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public int NoteId { get; set; }
    public Note? Note { get; set; }
}
